File: Managed/UnrealSharp/UnrealSharp/Interop/UWorldExporter.cs
using UnrealSharp.Binds;
using UnrealSharp.Core;
using UnrealSharp.CoreUObject;
using UnrealSharp.Engine;

namespace UnrealSharp.Interop;

[NativeCallbacks]
public static unsafe partial class UWorldExporter
{
    public static delegate* unmanaged<IntPtr, FName, float, NativeBool, float, FTimerHandle*, void> SetTimer;
    public static delegate* unmanaged<IntPtr, FTimerHandle*, void> InvalidateTimer;
    public static delegate* unmanaged<IntPtr, IntPtr, IntPtr> GetWorldSubsystem;
    public static delegate* unmanaged<IntPtr, IntPtr> GetNetMode;
}

File: Source/UnrealSharpCore/Export/UWorldExporter.cpp
#include "UWorldExporter.h"
#include "UnrealSharpCore/CSManager.h"
#include "Kismet/KismetSystemLibrary.h"

void UUWorldExporter::SetTimer(UObject* Object, FName FunctionName, float Rate, bool Loop, float InitialDelay, FTimerHandle* TimerHandle)
{
	FTimerDynamicDelegate Delegate;
	Delegate.BindUFunction(Object, FunctionName);
	*TimerHandle = UKismetSystemLibrary::K2_SetTimerDelegate(Delegate, Rate, Loop, false, InitialDelay);
}

void UUWorldExporter::InvalidateTimer(UObject* Object, FTimerHandle* TimerHandle)
{
	if (!IsValid(Object))
	{
		return;
	}

	Object->GetWorld()->GetTimerManager().ClearTimer(*TimerHandle);
}

void* UUWorldExporter::GetWorldSubsystem(UClass* SubsystemClass, UObject* WorldContextObject)
{
	if (!IsValid(WorldContextObject))
	{
		return nullptr;
	}
	
	UWorldSubsystem* WorldSubsystem = WorldContextObject->GetWorld()->GetSubsystemBase(SubsystemClass);
	return UCSManager::Get().FindManagedObject(WorldSubsystem);
}

void* UUWorldExporter::GetNetMode(UObject* WorldContextObject)
{
	if (!IsValid(WorldContextObject))
	{
		return nullptr;
	}
	return (void*)WorldContextObject->GetWorld()->GetNetMode();
}
